File: backend/src/main/java/com/backend/ord/seeders/entities/UserSeeder.java
package com.backend.ord.seeders.entities;

import com.backend.ord.domain.entities.User;
import com.backend.ord.repositories.UserRepository;
import com.backend.ord.seeders.factories.UserMockFactory;
import org.springframework.stereotype.Component;

@Component
public class UserSeeder implements SeederInterface<User> {
    private final UserRepository userRepository;
    private final UserMockFactory userMockFactory;

    public UserSeeder(UserRepository userRepository, UserMockFactory userMockFactory) {
        this.userRepository = userRepository;
        this.userMockFactory = userMockFactory;
    }

    @Override
    public User insertRow() {
        return userRepository.save(userMockFactory.mockEntity());
    }

    public User insertRowWithCredentials(String email, String password) {
        return userRepository.save(userMockFactory.mockEntityWithCredentials(email, password));
    }

    @Override
    public void deleteAll() {
        userRepository.deleteAll();
    }
}

File: backend/src/main/kotlin/com/backend/ord/seeders/entities/UserSeeder.kt
package com.backend.ord.seeders.entities

import com.backend.ord.core.user.UserRepository
import com.backend.ord.core.user.model.UserEntity
import com.backend.ord.seeders.factories.UserMockFactory
import org.springframework.stereotype.Component

@Component
class UserSeeder(
    private val userRepository: UserRepository,
    private val userMockFactory: UserMockFactory
) : SeederInterface<UserEntity> {
    override fun seedOneEntity(data: UserEntity?): UserEntity {
        return userRepository.save(data ?: userMockFactory.mockEntity())
    }

    fun insertRowWithCredentials(email: String, password: String): UserEntity {
        return userRepository.save(
            userMockFactory.mockEntity(
                email = email,
                password = password
            )
        )
    }

    override fun deleteAll() {
        userRepository.deleteAll()
    }
}
